osgp/platform/osgp-adapter-ws-publiclighting/src/main/java/org/opensmartgridplatform/adapter/ws/publiclighting/infra/jms/messageprocessor/DomainResponseMessageProcessor.java

```diff
@@ -0,0 +1,148 @@
+
+import java.io.Serializable;
+
+import javax.annotation.PostConstruct;
+import javax.jms.JMSException;
+import javax.jms.ObjectMessage;
+
+import org.opensmartgridplatform.adapter.ws.domain.entities.ResponseData;
+import org.opensmartgridplatform.adapter.ws.schema.core.notification.NotificationType;
+import org.opensmartgridplatform.adapter.ws.shared.services.NotificationService;
+import org.opensmartgridplatform.adapter.ws.shared.services.ResponseDataService;
+import org.opensmartgridplatform.shared.infra.jms.Constants;
+import org.opensmartgridplatform.shared.infra.jms.CorrelationIds;
+import org.opensmartgridplatform.shared.infra.jms.MessageProcessor;
+import org.opensmartgridplatform.shared.infra.jms.MessageProcessorMap;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+import org.opensmartgridplatform.shared.infra.jms.ResponseMessageResultType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+/**
+ * Base class for MessageProcessor implementations. Each MessageProcessor
+ * implementation should be annotated with @Component. Further the MessageType
+ * the MessageProcessor implementation can process should be passed in at
+ * construction. The Singleton instance is added to the HashMap of
+ * MessageProcessors after dependency injection has completed.
+ */
+public class DomainResponseMessageProcessor implements MessageProcessor {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DomainResponseMessageProcessor.class);
+
+    @Qualifier("domainPublicLightingResponseMessageProcessorMap")
+    @Autowired
+    protected MessageProcessorMap domainResponseMessageProcessorMap;
+
+    @Autowired
+    private NotificationService notificationService;
+
+    @Autowired
+    private ResponseDataService responseDataService;
+
+    /**
+     * The message type that a message processor implementation can handle.
+     */
+    protected MessageType messageType;
+
+    /**
+     * Construct a message processor instance by passing in the message type.
+     *
+     * @param messageType
+     *            The message type a message processor can handle.
+     */
+    protected DomainResponseMessageProcessor(final MessageType messageType) {
+        this.messageType = messageType;
+    }
+
+    /**
+     * Initialization function executed after dependency injection has finished.
+     * The MessageProcessor Singleton is added to the HashMap of
+     * MessageProcessors.
+     */
+    @PostConstruct
+    public void init() {
+        this.domainResponseMessageProcessorMap.addMessageProcessor(this.messageType, this);
+    }
+
+    @Override
+    public void processMessage(final ObjectMessage message) throws JMSException {
+        LOGGER.debug("Processing smart metering response message");
+
+        String correlationUid = null;
+        String messageType = null;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Rename "messageType" which hides the field declared at line 55. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://ci.opensmartgridplatform.org/sonarqube/coding_rules#rule_key=squid%3AHiddenFieldCheck)